openpype/hosts/maya/api/lib_rendersettings.py

```diff
@@ -0,0 +1,219 @@
+# -*- coding: utf-8 -*-
+"""Class for handling Render Settings."""
+from maya import cmds  # noqa
+import maya.mel as mel
+import six
+import sys
+
+from openpype.api import (
+    get_project_settings,
+    get_asset)
+
+from openpype.pipeline import legacy_io
+from openpype.pipeline import CreatorError
+
+
+class RenderSettings(object):
+
+    _image_prefix_nodes = {
+        'mentalray': 'defaultRenderGlobals.imageFilePrefix',
+        'vray': 'vraySettings.fileNamePrefix',
+        'arnold': 'defaultRenderGlobals.imageFilePrefix',
+        'renderman': 'defaultRenderGlobals.imageFilePrefix',
+        'redshift': 'defaultRenderGlobals.imageFilePrefix'
+    }
+
+    _image_prefixes = {
+        'mentalray': 'maya/<Scene>/<RenderLayer>/<RenderLayer>{aov_separator}<RenderPass>',  # noqa
+        'vray': 'maya/<scene>/<Layer>/<Layer>',
+        'arnold': 'maya/<Scene>/<RenderLayer>/<RenderLayer>{aov_separator}<RenderPass>',  # noqa
+        'renderman': 'maya/<Scene>/<layer>/<layer>{aov_separator}<aov>',
+        'redshift': 'maya/<Scene>/<RenderLayer>/<RenderLayer>{aov_separator}<RenderPass>'  # noqa
+    }
+
+    _aov_chars = {
+        "dot": ".",
+        "dash": "-",
+        "underscore": "_"
+    }
+
+    @classmethod
+    def get_image_prefix_attr(cls, renderer):
+        return cls._image_prefix_nodes[renderer]
+
+    def __init__(self, project_settings=None):
+        self._project_settings = project_settings
+        if not self._project_settings:
+            self._project_settings = get_project_settings(
+                legacy_io.Session["AVALON_PROJECT"]
+            )
+
+    @staticmethod
+    def apply_defaults(renderer=None, project_settings=None):
+        if renderer is None:
+            renderer = cmds.getAttr(
+                'defaultRenderGlobals.currentRenderer').lower()
```

Comment:
This does assume the renderer isn't overridden for a specific layer since the RenderSettings don't seem to trigger against a certain Maya layer.